DAO/src/lib.rs

```diff
@@ -0,0 +1,698 @@
+                &self.approved_token_program_id,
+                &proposal.applicant,
+                proposal.amount,
+            )
+            .await;
+        }
+        msg::reply(
+            DaoEvent::ProcessProposal {
+                applicant: proposal.applicant,
+                proposal_id,
+                did_pass: proposal.did_pass,
+            },
+            exec::gas_available() - GAS_RESERVE,
+            0,
+        );
+        self.proposals.insert(proposal_id, proposal);
+    }
+
+    // Withdraws the capital of the member
+    // Requirements:
+    // * `msg::source()` must be DAO member
+    // * The member must have sufficient amount
+    // * The latest proposal the member voted YES must be processed
+    // * Admin can ragequit only after transferring his role to another actor
+    // Arguments:
+    // * `amount`: The amount of shares the member would like to withdraw (the shares are converted to ERC20 tokens)
+    async fn ragequit(&mut self, amount: u128) {
+        if self.admin == msg::source() {
+            panic!("admin can not ragequit");
+        }
+        if !self.members.contains_key(&msg::source()) {
+            panic!("account is not a DAO member");
+        }
+        let member = self.members.get_mut(&msg::source()).unwrap();
+        if amount > member.shares {
+            panic!("unsufficient shares");
+        }
+
+        let proposal_id = member.highest_index_yes_vote;
+        if !self.proposals.get(&proposal_id).unwrap().processed {
+            panic!("cant ragequit until highest index proposal member voted YES on is processed");
+        }
+        member.shares = member.shares.saturating_sub(amount);
+        let funds = self.redeemable_funds(amount).await;
+        transfer_tokens(&self.approved_token_program_id, &msg::source(), funds).await;
+        self.total_shares = self.total_shares.saturating_sub(amount);
+        msg::reply(
+            DaoEvent::RageQuit {
+                member: msg::source(),
+                amount: funds,
+            },
+            exec::gas_available() - GAS_RESERVE,
+            0,
+        );
+    }
+
+    // Cancels the proposal after the end of the voting period if there are no YES votes.
+    // Requirements:
+    // * `msg::source()` must be the proposer
+    // * It can be cancelled if the number of YES votes is less than number of NO votes or the required quorum is not achieved
+    // * The voting period must be over
+    // * The proposal must not be cancelled or aborted
+    // Arguments:
+    // * `proposal_id`: the proposal ID
+    async fn cancel_proposal(&mut self, proposal_id: u128) {
+        if !self.proposals.contains_key(&proposal_id) {
+            panic!("proposal does not exist");
+        }
+        if self.proposals.get(&proposal_id).unwrap().proposer != msg::source() {
+            panic!("caller must be proposer");
+        }
+        let proposal = self.proposals.get_mut(&proposal_id).unwrap();
```

Comment:
this can be moved to one line above and avoid `self.proposals.get()`

Reply:
I need a mutable reference